src/app/components/challenge/challengeleaderboard/challengeleaderboard.component.ts

```diff
@@ -245,6 +260,9 @@ export class ChallengeleaderboardComponent implements OnInit, AfterViewInit {
       SELF.selectedPhaseSplit = phaseSplit;
       if (SELF.selectedPhaseSplit) {
         SELF.fetchLeaderboard(SELF.selectedPhaseSplit['id']);
+        if (SELF.phaseSplits[SELF.selectedPhaseSplit['id']].hasOwnProperty('show_leaderboard_by_latest_submission')) {
+          SELF.showLeaderboardByLatest = SELF.selectedPhaseSplit.show_leaderboard_by_latest_submission;
```

Comment:
Hey @drepram I've tested the field is showing on my local and there's not gonna be any case for this field to be missing so you can remove these if conditions and also you should initialize here the text variable as I've said on the above comment. So total change on this section should be like

`SELF.showLeaderboardByLatest = SELF.selectedPhaseSplit.show_leaderboard_by_latest_submission;`
`SELF.sortLeaderboardTextOption = SELF.showLeaderboardByLatest ? 'Sort by best' : 'Sort by latest';`

Reply:
Changed!